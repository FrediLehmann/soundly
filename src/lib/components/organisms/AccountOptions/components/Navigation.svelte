<script lang="ts" context="module">
  import { Settings, Upload } from '$lib/components/Icons';

  import NavLink from './NavLink.svelte';
</script>

<div
  class="shadow-md divide-y divide-gray-300 divide-solid absolute ml-5 top-11 border border-gray-200">
  <NavLink href="/upload"><Upload class="w-5 h-5" /> Upload</NavLink>
  <NavLink href="/uploads"><Upload class="w-5 h-5" /> Uploads</NavLink>
  <NavLink href="/profile"><Settings class="w-5 h-5" />Settings</NavLink>
</div>
