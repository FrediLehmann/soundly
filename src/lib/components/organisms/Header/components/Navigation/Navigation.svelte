<nav>
  <ul>
    <li>
      <a href="/upload" sveltekit:prefetch>Upload</a>
    </li>
    <li>
      <a href="/about" sveltekit:prefetch>About</a>
    </li>
    <li>
      <a href="/api" sveltekit:prefetch>API</a>
    </li>
  </ul>
</nav>

<style>
  nav {
    margin-inline-start: auto;
    background: hsla(var(--primary))
      linear-gradient(
        20deg,
        hsla(var(--primary)) 20%,
        hsla(var(--secondary)) 80%
      );
  }

  ul {
    list-style: none;
    display: flex;

    padding: 0;
    margin: 0;
  }

  li {
    position: relative;
    border-right: 4px hsla(var(--neutral-000)) solid;
  }

  li::before {
    content: '';
    position: absolute;
    background-color: hsla(var(--neutral-000));
    width: 100%;
    height: calc(100% - 5px);
    z-index: 1;
  }

  li:hover::before {
    opacity: 0.6;
  }

  a {
    position: relative;
    text-decoration: none;
    color: hsla(var(--neutral-900));
    display: inline-block;
    padding: 0.5rem 1rem;
    z-index: 2;
  }
</style>
