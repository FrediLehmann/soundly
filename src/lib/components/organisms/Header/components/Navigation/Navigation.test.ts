/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import Navigation from './Navigation.svelte';

describe('<Navigation /> component', () => {
  test('renders correctly', () => {
    const { component } = render(Navigation);
    expect(component).toMatchSnapshot();
  });
});
