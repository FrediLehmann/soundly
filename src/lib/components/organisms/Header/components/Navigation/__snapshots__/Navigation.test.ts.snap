// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> component renders correctly 1`] = `
Navigation {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {},
    "root": <div>
      <nav
        class="svelte-1upcm0y"
      >
        <ul
          class="svelte-1upcm0y"
        >
          <li
            class="svelte-1upcm0y"
          >
            <a
              class="svelte-1upcm0y"
              href="/upload"
              sveltekit:prefetch=""
            >
              Upload
            </a>
          </li>
           
          <li
            class="svelte-1upcm0y"
          >
            <a
              class="svelte-1upcm0y"
              href="/about"
              sveltekit:prefetch=""
            >
              About
            </a>
          </li>
           
          <li
            class="svelte-1upcm0y"
          >
            <a
              class="svelte-1upcm0y"
              href="/api"
              sveltekit:prefetch=""
            >
              API
            </a>
          </li>
        </ul>
      </nav>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
}
`;
